src/utils/network_calculations/ybus_calculations.jl

```diff
@@ -141,39 +121,75 @@ function _buildybus(branches, nodes, fixed_admittances)
         _ybus!(ybus, b, num_bus)
 
     end
-
     for fa in fixed_admittances
         _ybus!(ybus, fa, num_bus)
     end
+    return SparseArrays.dropzeros!(ybus)
+end
 
-    return ybus
-
+function _goderya(ybus::SparseArrays.SparseMatrixCSC{ComplexF64, Int64})
+    node_count = size(ybus)[1]
+    max_I = node_count^2
+    I, J, val = SparseArrays.findnz(ybus)
+    T = SparseArrays.sparse(I, J, ones(length(val)))
+    T_ = T * T
+    for n in 1:(node_count - 1)
+        I, _, _ = SparseArrays.findnz(T_)
+        if length(I) == max_I
+            @info "The System has no islands"
+            break
+        elseif length(I) < max_I
+            temp = T_ * T
+            I_temp, _, _ = SparseArrays.findnz(temp)
+            if all(I_temp == I)
+                error("The system contains islands")
```

Comment:
Why do we want to give errors if the system contains islands? It is possible to solve Power Flows in islanded systems right? It is definitely a problem for Dynamic Simulations though, but we can use that in some way to stop the simulation. Any issues in PSI?

Reply:
If the system has islands the Jacobian is singular.

Reply:
I see. Better not support islanded systems for now.